Add release notes for 0.292 (#24825)
prestodb/presto: presto-docs/src/main/sphinx/release/release-0.292.rst

* Add logic to iceberg type converter for timestamp with timezone :pr:`23534`. `#23534 <https://github.com/prestodb/presto/pull/23534>`_
* Add manifest file caching for deployments which use the Hive metastore. `#24481 <https://github.com/prestodb/presto/pull/24481>`_
* Add support for the ``hive.affinity-scheduling-file-section-size`` configuration property and ``affinity_scheduling_file_section_size`` session property. `#24598 <https://github.com/prestodb/presto/pull/24598>`_
* Add support of ``renaming table`` for Iceberg connector when configured with ``HIVE`` file catalog. `#24312 <https://github.com/prestodb/presto/pull/24312>`_
* Add table properties ``write.data.path`` to specify independent data write paths for Iceberg tables. `#24397 <https://github.com/prestodb/presto/pull/24397>`_
* Deprecate some table property names in favor of property names from the Iceberg library. See :doc:`connector/iceberg.rst`. `#24581 <https://github.com/prestodb/presto/pull/24581>`_

> * Enable manifest caching by default. `#24481 <https://github.com/prestodb/presto/pull/24481>`_

Reviewer: I don't think this one makes sense. Why are we putting the word "Add" at the front here? This PR is changing the default of an existing configuration, so I don't think it fits

Kudu Connector Changes
______________________
* Replaced return type of beginDelete. `#24528 <https://github.com/prestodb/presto/pull/24528>`_

Tpc-ds Connector Changes
________________________
* Add config property ``tpcds.use-varchar-type`` to allow toggling of char columns to varchar columns. `#24406 <https://github.com/prestodb/presto/pull/24406>`_

SPI Changes
___________
* Add ConnectorSession as an argument to PlanChecker.validate and PlanChecker.validateFragment. `#24557 <https://github.com/prestodb/presto/pull/24557>`_
* Add DeleteTableHandle support these changes in Metadata. `#24528 <https://github.com/prestodb/presto/pull/24528>`_
* Add ``CoordinatorPlugin#getExpressionOptimizerFactories`` to customize expression evaluation in the Presto coordinator. :pr:`24144`. `#24144 <https://github.com/prestodb/presto/pull/24144>`_
* Add a separate ConnectorDeleteTableHandle interface for `ConnectorMetadata.beginDelete` and `ConnectorMetadata.finishDelete`, replacing the previous usage of ConnectorTableHandle. `#24528 <https://github.com/prestodb/presto/pull/24528>`_
* Move IndexSourceNode to the SPI. `#24678 <https://github.com/prestodb/presto/pull/24678>`_

> * REMOTE_BUFFER_CLOSE_FAILED is now a retriable error. `#24808 <https://github.com/prestodb/presto/pull/24808>`_

Reviewer: Not sure if "improve" is correct here. We are changing the property of an existing error code. And whether it is "improving" anything is up to interpretation by the user

> * REMOTE_BUFFER_CLOSE_FAILED is now a retriable error. `#24808 <https://github.com/prestodb/presto/pull/24808>`_

Reviewer 2: Good point! (rereads original PR)

```suggestion
* Fix query failures by setting ``REMOTE_BUFFER_CLOSE_FAILED`` as a retriable error. `#24808 <https://github.com/prestodb/presto/pull/24808>`_
```

Elastic Search Changes
______________________
* Improve cryptographic protocol in response to `java:S4423 <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. `#24474 <https://github.com/prestodb/presto/pull/24474>`_

Iceberg Changes
_______________
* Iceberg connector support for ``UPDATE`` SQL statements. `#24281 <https://github.com/prestodb/presto/pull/24281>`_

Prestissimo (native Execution) Changes
______________________________________

> * Add Presto native shared arbitrator configuration properties:. `#24720 <https://github.com/prestodb/presto/pull/24720>`_

Reviewer 2: Because line 113 should be grouped with lines 115-120, I suggest moving lines 115-120 as a nested list underneath this line, and deleting the repetitions of the PR link to leave only one PR link at the end. 

Here's screenshots of how it looks in a local doc build, and what the code looks like in Visual Studio: 

local doc build
<img width="711" alt="Screenshot 2025-03-28 at 4 00 52 PM" src="https://github.com/user-attachments/assets/97caf75b-e050-4cc5-8614-697d73a833af" />

Visual Studio
<img width="675" alt="Screenshot 2025-03-28 at 4 00 38 PM" src="https://github.com/user-attachments/assets/fef5d6c0-e474-44cb-a80a-a01cd7e8b021" />

* Add module presto-native-tests to run end-to-end tests with Presto native workers. `#24234 <https://github.com/prestodb/presto/pull/24234>`_
* Add map of node ID to plan node to QueryCompletedEvent in the event listener interface. `#24590 <https://github.com/prestodb/presto/pull/24590>`_
* Add support for multiple query event listeners. `#24456 <https://github.com/prestodb/presto/pull/24456>`_
* Add ``spark.dynamic-presto-memory-pool-tuning-enabled`` configuration property to dynamically configure available Spark executor memory based on available container memory. `#24714 <https://github.com/prestodb/presto/pull/24714>`_
* Add the ability to canonicalize JSON output through session property ``canonicalized_json_extract``. `#24614 <https://github.com/prestodb/presto/pull/24614>`_
* Remove org.apache.logging.log4j:log4j-api from root POM. `#24605 <https://github.com/prestodb/presto/pull/24605>`_
* Remove org.apache.logging.log4j:log4j-core from root POM. `#24605 <https://github.com/prestodb/presto/pull/24605>`_

> * Enable file-based hive metastore to use HDFS/S3 location as warehouse dir. `#24660 <https://github.com/prestodb/presto/pull/24660>`_

Reviewer 2: ```suggestion
* Add the ability of a file-based Hive metastore to use HDFS/S3 location as warehouse dir. `#24660 <https://github.com/prestodb/presto/pull/24660>`_
```

Please move this to group with the other Add entries.

> * Enable node pool type specification when reporting to the coordinator from a C++ worker. `#24569 <https://github.com/prestodb/presto/pull/24569>`_

Reviewer 2: ```suggestion
* Add node pool type specification when reporting to the coordinator from a C++ worker. `#24569 <https://github.com/prestodb/presto/pull/24569>`_
```
Reading the original PR, I think this entry should be moved to the Presto C++ topic.

* Upgrade bootstrap to version 5. `#24167 <https://github.com/prestodb/presto/pull/24167>`_
* Upgrade jQuery to version 3.7.1. `#24167 <https://github.com/prestodb/presto/pull/24167>`_

> * Improve how multiple operator stats are merged together. `#24414 <https://github.com/prestodb/presto/pull/24414>`_

Reviewer 2: Please move these three Improve entries in lines 44-46 to the other Improve entries in the General Changes topic.

* Upgrade org.apache.calcite to 1.38.0 in response to `CVE-2023-2976 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-2976>`_. `#24706 <https://github.com/prestodb/presto/pull/24706>`_
* Upgrade org.apache.ratis to 3.1.3 in response to `CVE-2020-15250 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15250>`_. `#24496 <https://github.com/prestodb/presto/pull/24496>`_
* Upgrade aws-java-sdk version to 1.12.782 in response to `CVE-2024-21634 <https://nvd.nist.gov/vuln/detail/cve-2024-21634>`_. `#24606 <https://github.com/prestodb/presto/pull/24606>`_
* Upgrade json-smart version to 2.5.2 in response to `CVE-2024-57699 <https://nvd.nist.gov/vuln/detail/CVE-2024-57699>`_. `#24631 <https://github.com/prestodb/presto/pull/24631>`_
* Upgrade the accumulo version to 1.10.1 in response to `CVE-2020-17533 <https://github.com/advisories/GHSA-grc3-8q8m-4j7c>`_. `#24438 <https://github.com/prestodb/presto/pull/24438>`_
* Upgrade the hive-dwrf version to 0.8.7 which involved upgrading snappy version to 0.5 in response to `CVE-2024-36124 <https://github.com/advisories/GHSA-8wh2-6qhj-h7j9>`_. `#24461 <https://github.com/prestodb/presto/pull/24461>`_

> * Remove reload4j dependency in response to `WS-2022-0467 <https://www.mend.io/vulnerability-database/WS-2022-0467>`_. `#24606 <https://github.com/prestodb/presto/pull/24606>`_

Reviewer 2: Please move this Remove entry, and the Replace entry in line 88, to before the Upgrade entries in the Security topic.

TPC-DS Connector Changes
________________________
* Add configuration property ``tpcds.use-varchar-type`` to allow toggling of char columns to varchar columns. `#24406 <https://github.com/prestodb/presto/pull/24406>`_

SPI Changes
___________
* Add ConnectorSession as an argument to PlanChecker.validate and PlanChecker.validateFragment. `#24557 <https://github.com/prestodb/presto/pull/24557>`_
* Add DeleteTableHandle support for the ConnectorTableHandles changes in Metadata. `#24528 <https://github.com/prestodb/presto/pull/24528>`_
* Add ``CoordinatorPlugin#getExpressionOptimizerFactories`` to customize expression evaluation in the Presto coordinator. `#24144 <https://github.com/prestodb/presto/pull/24144>`_
* Add a separate ConnectorDeleteTableHandle interface for ``ConnectorMetadata.beginDelete`` and ``ConnectorMetadata.finishDelete``, replacing the previous usage of ConnectorTableHandle. `#24528 <https://github.com/prestodb/presto/pull/24528>`_
* Update ``beginDelete`` to return new types, and ``finishDelete`` to accept new types in ``ConnectorMetadata``. `#24528 <https://github.com/prestodb/presto/pull/24528>`_

> * Move IndexSourceNode to the SPI. `#24678 <https://github.com/prestodb/presto/pull/24678>`_

Reviewer 2: Please move these Move and Fix entries in lines 138 and 139 to the correct [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) sequence in the SPI Changes topic.

* Upgrade prismJs to 1.30.0 in response to `CVE-2024-53382 <https://github.com/advisories/GHSA-x7hr-w5r2-h6wg>`_. `#24765 <https://github.com/prestodb/presto/pull/24765>`_
* Upgrade the errorprone dependency from version 2.28.0 to 2.36.0. `#24475 <https://github.com/prestodb/presto/pull/24475>`_
* Upgrade the io.grpc library from version 1.68.0 to 1.70.0 in response to `CVE-2024-7254 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-7254>`_, `CVE-2020-8908 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-8908>`_. `#24475 <https://github.com/prestodb/presto/pull/24475>`_
* Upgrade org.apache.logging.log4j:log4j-api from 2.17.1 to 2.24.3 in response to `CVE-2024-47554 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47554>`_. `#24507 <https://github.com/prestodb/presto/pull/24507>`_
* Upgrade org.apache.logging.log4j:log4j-core from 2.17.1 to 2.24.3 in response to `CVE-2024-47554 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47554>`_. `#24507 <https://github.com/prestodb/presto/pull/24507>`_

> * Introduce ``spark.dynamic-presto-memory-pool-tuning-enabled`` to dynamically configure available Spark executor memory based on available container memory `#24714 <https://github.com/prestodb/presto/pull/24714>`_

Reviewer 2: ```suggestion
* Add ``spark.dynamic-presto-memory-pool-tuning-enabled`` configuration property to dynamically configure available Spark executor memory based on available container memory. `#24714 <https://github.com/prestodb/presto/pull/24714>`_
```

* Upgrade the accumulo version to 1.10.1 in response to `CVE-2020-17533 <https://github.com/advisories/GHSA-grc3-8q8m-4j7c>`_. `#24438 <https://github.com/prestodb/presto/pull/24438>`_
* Upgrade the hive-dwrf version to 0.8.7 which involved upgrading snappy version to 0.5 in response to `CVE-2024-36124 <https://github.com/advisories/GHSA-8wh2-6qhj-h7j9>`_. `#24461 <https://github.com/prestodb/presto/pull/24461>`_

Hive Connector Changes
______________________
* Fix Parquet read failing for nested Decimal types. `#24440 <https://github.com/prestodb/presto/pull/24440>`_
* Fix getting views for Hive metastore 2.3+. `#24466 <https://github.com/prestodb/presto/pull/24466>`_
* Add session property ``hive.stats_based_filter_reorder_disabled`` for disabling reader stats based filter reordering. `#24630 <https://github.com/prestodb/presto/pull/24630>`_
* Replace return type of beginDelete. `#24528 <https://github.com/prestodb/presto/pull/24528>`_
* Rename session property ``hive.stats_based_filter_reorder_disabled`` to ``hive.native_stats_based_filter_reorder_disabled``. `#24637 <https://github.com/prestodb/presto/pull/24637>`_

> * Updated native `HiveConnectorProtocol` to supply `NotImplemented` for `ConnectorDeleteTableHandle` type. `#24721 <https://github.com/prestodb/presto/pull/24721>`_

Reviewer 2: ```suggestion
* Update native ``HiveConnectorProtocol`` to supply ``NotImplemented`` for ``ConnectorDeleteTableHandle`` type. `#24721 <https://github.com/prestodb/presto/pull/24721>`_
```

* Add manifest file caching for deployments which use the Hive metastore. `#24481 <https://github.com/prestodb/presto/pull/24481>`_
* Add support for the ``hive.affinity-scheduling-file-section-size`` configuration property and ``affinity_scheduling_file_section_size`` session property. `#24598 <https://github.com/prestodb/presto/pull/24598>`_
* Add support of ``renaming table`` for Iceberg connector when configured with ``HIVE`` file catalog. `#24312 <https://github.com/prestodb/presto/pull/24312>`_
* Add table property ``write.data.path`` to specify independent data write paths for Iceberg tables. `#24397 <https://github.com/prestodb/presto/pull/24397>`_
* Deprecate some table property names in favor of property names from the Iceberg library. See :doc:`/connector/iceberg`. `#24581 <https://github.com/prestodb/presto/pull/24581>`_
* Enable manifest caching by default. `#24481 <https://github.com/prestodb/presto/pull/24481>`_
* Support for Iceberg table sort orders. Tables can be created to add a list of `sorted_by` columns which will be used to order files written to the table. `#21977 <https://github.com/prestodb/presto/pull/21977>`_

> * Iceberg connector support for ``UPDATE`` SQL statements. `#24281 <https://github.com/prestodb/presto/pull/24281>`_

Reviewer 2: ```suggestion
* Add Iceberg connector support for ``UPDATE`` SQL statements. `#24281 <https://github.com/prestodb/presto/pull/24281>`_
```

> * Updated native `IcebergConnectorProtocol` to supply `NotImplemented` for `ConnectorDeleteTableHandle` type. `#24721 <https://github.com/prestodb/presto/pull/24721>`_

Reviewer 2: ```suggestion
* Update native ``IcebergConnectorProtocol`` to supply ``NotImplemented`` for ``ConnectorDeleteTableHandle`` type. `#24721 <https://github.com/prestodb/presto/pull/24721>`_
```

Prestissimo (native Execution) Changes
______________________________________
* Add a native type manager. `#24179 <https://github.com/prestodb/presto/pull/24179>`_
* Add support for Apache Arrow Flight connectors `#24504 <https://github.com/prestodb/presto/pull/24504>`_
* Add Presto native shared arbitrator configuration properties:. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.global-arbitration-abort-time-ratio``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.global-arbitration-memory-reclaim-pct``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.global-arbitration-without-spill``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.memory-pool-abort-capacity-limit``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.memory-pool-min-reclaim-bytes``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_
* ``shared-arbitrator.memory-reclaim-threads-hw-multiplier``. `#24720 <https://github.com/prestodb/presto/pull/24720>`_

> * Set default value for ``task.max-drivers-per-task`` to use thread concurrency of the host `#24642 <>`

Reviewer 2: ```suggestion
* Add a default value for ``task.max-drivers-per-task`` to use thread concurrency of the host. `#24642 <https://github.com/prestodb/presto/pull/24642>`_
```

Reviewer: Not sure if "Add" makes sense here. A default value existed previously, so we're changing it, rather than "adding" something

> * Added a type parameter for `ConnectorDeleteTableHandle` implementations to `ConnectorProtocolTemplate`, along with support for (de)serialization of connector-specific types.  Existing native connector implementations defining `ConnectorProtocolTemplate` specializations will need to update their definitions to supply their specific type or use `NotImplemented`. `#24721 <https://github.com/prestodb/presto/pull/24721>`_

Reviewer 2: ```suggestion
* Add a type parameter for ``ConnectorDeleteTableHandle`` implementations to ``ConnectorProtocolTemplate``, along with support for (de)serialization of connector-specific types.  Existing native connector implementations defining ``ConnectorProtocolTemplate`` specializations must update their definitions to supply their specific type or use ``NotImplemented``. `#24721 <https://github.com/prestodb/presto/pull/24721>`_
```

* Add ``exchange.http-client.request-data-sizes-max-wait-sec`` to native system configs. `#24774 <https://github.com/prestodb/presto/pull/24774>`_
* Add ``spill-enabled``, ``join-spill-enabled``, ``aggregation-spill-enabled``, and ``order-by-spill-enabled`` to native system configs. `#24726 <https://github.com/prestodb/presto/pull/24726>`_

> * Added new error code name ``MEMORY_ARBITRATION_FAILURE`` under error code ``INSUFFICIENT_RESOURCE``. `#24773 <https://github.com/prestodb/presto/pull/24773>`_

Reviewer 2: ```suggestion
* Add new error code name ``MEMORY_ARBITRATION_FAILURE`` under error code ``INSUFFICIENT_RESOURCE``. `#24773 <https://github.com/prestodb/presto/pull/24773>`_
```

* Improve metrics creation by refactoring local variables to a dedicated class. `#24414 <https://github.com/prestodb/presto/pull/24414>`_
* Improve efficiency of coordinator when running a large number of tasks, controlled by ``task.enable-event-loop``. `#24668 <https://github.com/prestodb/presto/pull/24668>`_

Prestissimo (Native Execution) Changes
______________________________________
* Add a native type manager. `#24179 <https://github.com/prestodb/presto/pull/24179>`_
* Add support for Apache Arrow Flight connectors `#24504 <https://github.com/prestodb/presto/pull/24504>`_
* Add Presto native shared arbitrator configuration properties:
    * ``shared-arbitrator.global-arbitration-abort-time-ratio``.
    * ``shared-arbitrator.global-arbitration-memory-reclaim-pct``.
    * ``shared-arbitrator.global-arbitration-without-spill``.
    * ``shared-arbitrator.memory-pool-abort-capacity-limit``.
    * ``shared-arbitrator.memory-pool-min-reclaim-bytes``.

>     * ``shared-arbitrator.memory-reclaim-threads-hw-multiplier``.

Reviewer 2: ```suggestion
    * ``shared-arbitrator.memory-reclaim-threads-hw-multiplier``.
    
```
Need blank line after this or the error 
```
/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/release/release-0.292.rst:59: WARNING: Bullet list ends without a blank line; unexpected unindent. [docutils]
```
appears in the local doc build.
